src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/MSBuildItemIdentityComparer.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// A comparer for the <see cref="IMSBuildItem"/> based on the <see cref="IMSBuildItem.Identity"/>.
+    /// </summary>
+    internal class MSBuildItemIdentityComparer : IEqualityComparer<IMSBuildItem>
```

Comment:
Minor performance improvement: Create a readonly static instance, and where this class is used, use that static instance, rather than creating a new instance each time.